osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs

```diff
@@ -231,24 +145,24 @@ protected override void UpdateStartTimeStateTransforms()
         {
             base.UpdateStartTimeStateTransforms();
 
-            using (BeginDelayedSequence(-ring_appear_offset))
-                targetRing.ScaleTo(target_ring_scale, 400, Easing.OutQuint);
+            (spinnerBody.Drawable as ISkinnableSwell)?.ApplyPassiveTransforms(this, MainPiece);
         }
 
         protected override void UpdateHitStateTransforms(ArmedState state)
         {
-            const double transition_duration = 300;
-
             switch (state)
             {
                 case ArmedState.Idle:
-                    expandingRing.FadeTo(0);
+                    HandleUserInput = true;
                     break;
 
                 case ArmedState.Miss:
                 case ArmedState.Hit:
-                    this.FadeOut(transition_duration, Easing.Out);
-                    bodyContainer.ScaleTo(1.4f, transition_duration);
+                    // Postpone drawable hitobject expiration until it has animated/faded out. Inputs on the object are disallowed during this delay.
+                    LifetimeEnd = Time.Current + 1200;
```

Comment:
What is 1200 here?

Reply:
This is the maximum amount of time the completion animation can take (the `LegacySwell` animations uses this period fully). I will make this a constant because it's not clear indeed.